### ISSUE TITLE
Accessibility issues

### ISSUE THREAD

**Issue by ntbrown** [uikit/uikit#4261]
By setting a hard-coded font-size on the :root (html) element this fundamentally breaks accessibility by preventing font size scaling according to user preferences.

In the core css file(s) we have on the first block:

html {
font-size: 16px;
}

If you had set it to 1rem it'd be fine, or 100%, or even 62.5% if you were attempting to work in base 10. I'm assuming this was added to "enforce" a consistent 16px in every browser and rem styles based off of these, but failed to consider the accessibility issues here.

You can make arguments about how users "don't really do this". Regardless, it's a fools argument. Users do scale this way and in many cases the number of users that do is a larger market share than some browser usage(s). Even if they didn't it still fundamentally broken accessibility.

Throughout the file there are other notable accessibility issues. One of the more obvious ones being:

a:active,
a:hover {
  outline: none;
}

and similar outline adjustments. It's fundamentally recommended to _not_ do this unless you're replacing the outline style on every last one of these with a notable different state users can easily distinguish. It's a feature required for accessibility didn't bother checking all of these to see if that's true. But, I'm assuming not - people tend to fail that standard often. I'm sure there are more.

If I've missed some aspect of the framework that remedies these things happy to hear it, but at first glance there seem to be significant accessibility problems.

**Comment by gitgit2019** [comment]
I think as it's meant as a framework the responsibility is on users to write those accessibility standards in.
I do agree though on fontsize and margins, I wish the entire framework were in ems instead of pixels.

**Comment by ntbrown** [comment]
> I think as it's meant as a framework the responsibility is on users to write those accessibility standards in.
> I do agree though on fontsize and margins, I wish the entire framework were in ems instead of pixels.

Hardly. A framework that doesn't abide by accessibility standards is useless.

By writing those standards in you mean.... Rewriting the entire framework? At which point I fail to see why you'd bother in the first place.

No decent framework anyone uses does this.

Your ems comment is also wrong. Em is for very specific use cases (say small, medium, large buttons in response to font size change to scale them appropriately). Nesrly every thing else should be rem. Pixels only for well defined static elements that are idempotent.

**Comment by janschoenherr** [comment]
Thanks, version 3.16.0 comes with built in accessibility.
